src/data/set/basic.lean

```diff
@@ -716,6 +716,9 @@ theorem eq_of_mem_singleton {x y : α} (h : x ∈ ({y} : set α)) : x = y := h
 @[simp] theorem singleton_eq_singleton_iff {x y : α} : {x} = ({y} : set α) ↔ x = y :=
 ext_iff.trans eq_iff_eq_cancel_left
 
+lemma singleton_injective : injective (singleton : α → set α) :=
+λ _ _ h, singleton_eq_singleton_iff.mp h
```

Comment:
Does this work?
```suggestion
λ _ _, singleton_eq_singleton_iff.mp
```

Reply:
I think it did, but thought I saw the other style nearby